balance_check/providers/gamestop.py

```diff
@@ -0,0 +1,111 @@
+import sys
+import requests
+from bs4 import BeautifulSoup
+from balance_check import logger, captcha_solver, config
+from balance_check.providers import BalanceCheckProvider
+from balance_check.validators.gift_card import Merchant, GiftCardSchema
+
+
+class GameStop(BalanceCheckProvider):
+    def __init__(self):
+        super().__init__()
+
+        self.website_url = "https://www.gamestop.com/profiles/valuelookup.aspx"
+        self.schema = GiftCardSchema([Merchant.GameStop])
+
+    def scrape(self, fields):
+        session = requests.Session()
+        session.headers.update({
+            "User-Agent": config.USER_AGENT
+        })
+
+        fields["X-Requested-With"] = "XMLHttpRequest"
+
+        logger.info("Fetching balance check page")
+
+        resp = session.get(self.website_url)
+        if resp.status_code != 200:
+            logger.critical("Failed to GET website (status code {})".format(resp.status_code))
+            sys.exit(1)
+
+        page_html = BeautifulSoup(resp.content, features="html.parser")
+        form = page_html.find("form")
+        if not form:
+            logger.critical("Unable to find balance check form")
+            sys.exit(1)
+
+        endpoint = "{}{}".format(self.website_url, form['action'])
+
+        # These fields are present on GS balance check page, does not work without including them
+        fields['__EVENTTARGET'] = ''
+        fields['__EVENTARGUMENT'] = ''
+        fields['__LASTFOCUS'] = ''
+        fields['__VIEWSTATE'] = page_html.find("input",id='__VIEWSTATE')['value']
+        fields['__VIEWSTATEGENERATOR'] = page_html.find("input",id='__VIEWSTATEGENERATOR')['value']
+
+        recaptcha_field = page_html.find("div", class_="g-recaptcha")
+        if not recaptcha_field:
+            logger.critical("Unable to find reCAPTCHA")
+            sys.exit(1)
+
+        site_key = recaptcha_field["data-sitekey"]
+
+        logger.info("Solving reCAPTCHA (~30s)")
+
+        captcha_resp = captcha_solver.solve_recaptcha(self.website_url, site_key)
+        if captcha_resp["errorId"] != 0:
+            logger.critical("Unable to solve reCAPTCHA ({})".format(captcha_resp["errorDescription"]))
+            sys.exit(1)
+
+        fields["g-recaptcha-response"] = captcha_resp["solution"]["gRecaptchaResponse"]
+
+        logger.info("Fetching card balance")
+
+        session.headers.update({
```

Comment:
You may not need all of these headers... perhaps remove some and see if it still works, just to keep this minimal.

Reply:
Roger that